src/tree.rs

```diff
@@ -1126,7 +1126,9 @@ fn detect_cycles(parents: &[ResolvedParent]) -> Option<Index> {
         let mut grandparent_index = parent.index().and_then(|index| parents[index].index());
         while let (Some(i), Some(j)) = (parent_index, grandparent_index) {
             if i == j {
+                break_cycles(builder, i).expect("Can't break cycles");
                 return Some(i);
+                // return None;
```

Comment:
Just a fun fact: if I continue to not use the returned index here, I'll probably change the return type of detect_cycles. Let me know if that seems right.